internal/codec.go

```diff
@@ -110,6 +112,22 @@ func (j *jsonEncoder) Encode(msg proto.Message) error {
 	return nil
 }
 
+// This function is lifted from connect-go since this is the logic it uses
```

Comment:
It doesn't feel right to duplicate this logic here either. I think ideally it would be nice if the same logic that the Go implementation was using to encode was made available, but that doesn't feel right either. Let me know if there's a better way to handle this.

Reply:
I don't see any other way right now, so this seems fine.

It's a pet peave of mine that the connect-go library doesn't expose any of this (codec impls) for re-use. Ideally, it would just make its codec and compression impls public so you can wrap/re-use. 🤷